crates/unleash-edge-auth/src/token_validator.rs

```diff
@@ -46,6 +47,46 @@ impl TokenRegister for TokenValidator {
     }
 }
 
+pub fn create_revalidation_task(
+    validator: &Arc<TokenValidator>,
+    validation_interval_seconds: u64,
+) -> Pin<Box<dyn Future<Output = ()> + Send>> {
+    let validator_clone = validator.clone();
+    Box::pin(async move {
```

Comment:
`schedule_validation_of_known_tokens` becomes this function now. It's no longer part of TokenValidator so we can call it nicely but it's ultimately the same function just wrapped in Box::pin(async move{}) to make it returnable without executing it